libs/entitlement/asm-provider/README.md

```diff
@@ -0,0 +1,2 @@
+This module uses the ASM library to implement various things, including bytecode instrumentation.
```

Comment:
Would it ever make sense to rename `asm-provider` to `instrumentation`?

Reply:
Oh the irony.

That was the original name. @rjernst suggested generalizing the name because we anticipate we'll need ASM for more than just instrumentation. I think he said you had a need for ASM?

I think, for example, we'll want to use ASM (not reflection) to scan the `EntitlementChecker` class to discover which methods to instrument. (Using reflection would force a bunch of classes to be loaded, only for them to be retransformed and reloaded again; using ASM should make startup more efficient.)

Reply:
Haha, well I'm not going to be too strongly opinionated on names so carry on :)